### PR TITLE
Fix handling of null partition values

### PR DESCRIPTION
**Pull request by rdblue**

### CODE REVIEW

File: spark/src/test/java/com/netflix/iceberg/spark/source/TestPartitionValues.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+          .save(location.toString());
+
+      Dataset<Row> result = spark.read()
+          .format("iceberg")
+          .load(location.toString());
+
+      List<SimpleRecord> actual = result.orderBy("id").as(Encoders.bean(SimpleRecord.class)).collectAsList();
+
+      Assert.assertEquals("Number of rows should match", expected.size(), actual.size());
+      Assert.assertEquals("Result rows should match", expected, actual);
```
Comment (fbocse) [inline spark/src/test/java/com/netflix/iceberg/spark/source/TestPartitionValues.java]: Nit: given this assert would you consider the previous one redundant?

File: spark/src/main/java/com/netflix/iceberg/spark/source/Reader.java [hunk 2]
```diff
@@ -463,7 +463,12 @@ private static UnsafeProjection projection(Schema finalSchema, Schema readSchema
     @Override
     public InternalRow apply(StructLike tuple) {
       for (int i = 0; i < types.length; i += 1) {
-        reusedRow.update(i, convert(tuple.get(positions[i], javaTypes[i]), types[i]));
+        Object value = tuple.get(positions[i], javaTypes[i]);
+        if (value != null) {
+          reusedRow.update(i, convert(value, types[i]));
+        } else {
+          reusedRow.setNullAt(i);
```
Comment (fbocse) [inline spark/src/main/java/com/netflix/iceberg/spark/source/Reader.java]: 👍

File: spark/src/test/java/com/netflix/iceberg/spark/source/TestPartitionValues.java (hunk 1 above)
Comment (rdblue) [reply spark/src/test/java/com/netflix/iceberg/spark/source/TestPartitionValues.java]: This is to make failures easier to understand. If the number of rows is off, the second assertion will fail, but it is hard to read. The important piece of information for that failure case is that the number is incorrect.

**Comment by rdblue** [comment]
Thanks for reviewing, @fbocse!

**Pull request merged**